DXRModules/DeusEx/Classes/DXRFashion.uc

```diff
@@ -0,0 +1,385 @@
+}
+
+function class<ScriptedPawn> RandomCoatInfluencer()
+{
+    return coatInfluencers[Rand(numCoatInfluencers)];
+}
+
+function class<ScriptedPawn> RandomNonCoatInfluencer()
+{
+    return nonCoatInfluencers[Rand(numNonCoatInfluencers)];
+}
+
+function bool IsTrenchInfluencer(class<ScriptedPawn> influencer)
+{
+    return (influencer.Default.Mesh == LodMesh'DeusExCharacters.GM_Trench' ||
+            influencer.Default.Mesh == LodMesh'DeusExCharacters.GM_Trench_F');
+}
+
+function ApplyOutfit(Actor p, texture coat1, texture coat2, texture shirt, texture pants, bool isJC) {
+    local bool isTrench;
+    
+    isTrench = (coat1 != None && coat2 != None);
+    
+    if (isTrench) {
+        p.Mesh = LodMesh'DeusExCharacters.GM_Trench';
+        p.MultiSkins[1] = coat1;
+	    p.MultiSkins[5] = coat2;
+    	p.MultiSkins[4] = shirt;
+        p.MultiSkins[2] = pants;   
+    } else {
+        p.Mesh = LodMesh'DeusExCharacters.GM_DressShirt';
+    	p.MultiSkins[5] = shirt;
+        p.MultiSkins[3] = pants;
+        p.MultiSkins[1] = Texture'DeusExItems.Skins.PinkMaskTex';
+        p.MultiSkins[2] = Texture'DeusExItems.Skins.PinkMaskTex';
+    }
+    
+    if (isJC) {
+        p.MultiSkins[6] = Texture'DeusExCharacters.Skins.FramesTex4';
+        p.MultiSkins[7] = Texture'DeusExCharacters.Skins.LensesTex5';
+    } else {
+        p.MultiSkins[6] = Texture'DeusExItems.Skins.GrayMaskTex';
+        p.MultiSkins[7] = Texture'DeusExItems.Skins.BlackMaskTex';
+    }
+
+
+}
+
+//Brothers gotta match, mom got their clothes out in advance
+function GetDressed()
+{
+	local PaulDenton paul;
+	local PaulDentonCarcass paulCarcass;
+	local JCDentonMaleCarcass jcCarcass;
+	local JCDouble jc;  
+    local texture coat1,coat2,pants,shirt;
+    local name coatinfluencer,pantsinfluencer,shirtinfluencer;
+    local class<ScriptedPawn> styleInfluencer;
+    local bool isTrench;
+
+    coatinfluencer = dxr.Player.FlagBase.GetName('DXRFashion_CoatInfluencer');
+    pantsinfluencer = dxr.Player.FlagBase.GetName('DXRFashion_PantsInfluencer');
+    shirtinfluencer = dxr.Player.FlagBase.GetName('DXRFashion_ShirtInfluencer');
+
+    if (coatinfluencer == '' ||
+        pantsinfluencer == '' ||
+        shirtinfluencer == '') {
+        //This was probably a game saved before fashion existed
+        info("No stored outfit!");
+        InitInfluencers();
+        RandomizeClothes();
+        coatinfluencer = dxr.Player.FlagBase.GetName('DXRFashion_CoatInfluencer');
+        pantsinfluencer = dxr.Player.FlagBase.GetName('DXRFashion_PantsInfluencer');
+        shirtinfluencer = dxr.Player.FlagBase.GetName('DXRFashion_ShirtInfluencer');        
+    }
+    
+
+    if (coatinfluencer!='') {
+        styleInfluencer = class<ScriptedPawn>(GetClassFromString(string(coatinfluencer),class'ScriptedPawn'));
+        isTrench = IsTrenchInfluencer(styleInfluencer);
+        if (isTrench) {
+            coat1=GetCoat1(styleInfluencer);
+            coat2=GetCoat2(styleInfluencer);
```

Comment:
is it not possible to mix 2 different coat skins?

Reply:
You could, but I'm fairly certain it would look really bad at least 99% of the time.  The two parts are the upper part of the coat and the bottom part of the coat.